crates/storage/src/blueprint/sparse.rs

```diff
@@ -484,237 +484,483 @@ where
     }
 }
 
-/// The macro that generates SMT storage tests for the table with [`crate::structured_storage::test::InMemoryStorage`].
 #[cfg(feature = "test-helpers")]
-#[macro_export]
-macro_rules! root_storage_tests {
-    ($table:ident, $metadata_table:ident, $current_key:expr, $foreign_key:expr, $generate_key:ident, $generate_value:ident) => {
-        paste::item! {
-        #[cfg(test)]
-        mod [< $table:snake _root_tests >] {
-            use super::*;
-            use $crate::{
-                structured_storage::test::InMemoryStorage,
-                transactional::WriteTransaction,
-                StorageAsMut,
-            };
-            use $crate::rand::{
-                rngs::StdRng,
-                SeedableRng,
+#[allow(warnings)]
+/// Test module for root storage tests.
+pub mod root_storage_tests {
+    use fuel_vm_private::{
+        fuel_merkle::sparse::{
+            self,
+            proof::Proof,
+            MerkleTree,
+            MerkleTreeKey,
+        },
+        fuel_storage::{
+            Mappable,
+            MerkleRoot,
+            StorageAsMut,
+        },
+    };
+    use rand::{
+        rngs::StdRng,
+        SeedableRng,
+    };
+
+    use crate::{
+        blueprint::BlueprintMutate,
+        codec::{
+            Decode,
+            Encode,
+            Encoder,
+        },
+        structured_storage::{
+            test::InMemoryStorage,
+            StructuredStorage,
+            TableWithBlueprint,
+        },
+        tables::merkle::SparseMerkleMetadata,
+        transactional::{
+            StorageTransaction,
+            WriteTransaction,
+        },
+    };
+
+    use super::{
+        PrimaryKey,
+        Sparse,
+    };
+
+    /// Provides root storage tests for SMT storage tables.
+    pub trait RootStorageTests: SmtTableWithBlueprint
```

Comment:
Why do we need a default implementation within the `RootStorageTests` trait when we could have a blanket implementation?

Reply:
Not sure I follow what you mean. Do you mean why we provide default implementations of the tests in the trait definition, as opposed to implementing them in the blanket implementation on line 890?

I think this is cleaner, since redefining the methods in the blanket implementation would cause us to have to repeat them additionally one time. And I don't see any benefit of doing it the other way around - although I have nothing against doing that way either.

Is this what you were referring to or did I misunderstand something?